### PR TITLE
Add support for linking to Artsy on a public Artwork

### PR DESCRIPTION
**Pull request by orta**
https://github.com/artsy/energy/issues/181 

Now at the end of the artwork section of an email, there is a "Open on Artsy." - with "Artsy" being clickable and will take you directly to the artwork.

### CODE REVIEW

File: Classes/Categories/UIDevice+Testing.m [hunk 1]
```diff
@@ -3,20 +3,9 @@
...
-    static dispatch_once_t oncePredicate;
-    dispatch_once(&oncePredicate, ^{
-        NSString *XCInjectBundle = [[[NSProcessInfo processInfo] environment] objectForKey:@"XCInjectBundle"];
-
-        ARRunningUnitTests = [XCInjectBundle hasSuffix:@".xctest"];
-    });
-
-    return ARRunningUnitTests;
+    return NSClassFromString(@"XCTestCase") != nil;
 }
```
Comment (orta) [inline Classes/Categories/UIDevice+Testing.m]: not really sure why this wasn't working, but it's now what we use on Eigen

File: Classes/Util/App/CoreDataManager.m [hunk 2]
```diff
@@ -140,7 +140,7 @@ + (NSManagedObjectContext *)stubbedManagedObjectContext
 
     NSError *error = nil;
     NSManagedObjectModel *model = [CoreDataManager managedObjectModel];
-    persistentStoreCoordinator = [[NSPersistentStoreCoordinator alloc] initWithManagedObjectModel:model];
+    NSPersistentStoreCoordinator *persistentStoreCoordinator = [[NSPersistentStoreCoordinator alloc] initWithManagedObjectModel:model];
```
Comment (orta) [inline Classes/Util/App/CoreDataManager.m]: this is in the stubbed MOC, so it should be fresh each time

File: Classes/Util/Emails/AREmailComposer.m [hunk 3]
```diff
@@ -117,7 +117,7 @@ - (void)generateAnalyticEvent
     NSDictionary *properties = @{
         @"artworks" : @(self.artworks.count),
         @"price" : @(self.options.priceType > 0),
-        @"using_Exact_pricing" : @(self.options.priceType == AREmailSettingsPriceTypeBackend),
+        @"using_exact_pricing" : @(self.options.priceType == AREmailSettingsPriceTypeBackend),
```
Comment (orta) [inline Classes/Util/Emails/AREmailComposer.m]: @anipetrov - spotted a typo in an analytics event property, would this change worry you?

File: Classes/Categories/UIDevice+Testing.m (hunk 1 above)
Comment (alloy) [inline Classes/Categories/UIDevice+Testing.m]: This env var is gone with Xcode 7.3.

Comment (orta) [inline Classes/Categories/UIDevice+Testing.m]: legit

**Comment by ArtsyMobile** [comment]
&nbsp; | 1 Error
    ------------- | ------------
        :no_entry_sign: | No CHANGELOG changes made
    

  

  

  


<p align="right" meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://github.com/KrauseFx/danger/">danger</a>
</p>

**Comment by alloy** [comment]
Looks good to me 👍 The analytics issue can be dealt with in a new PR, if that turns out to be an issue.

**Pull request merged**